apis/apps/v1alpha1/apicast_types.go

```diff
@@ -28,46 +28,81 @@ import (
 	// +optional
 	Replicas *int64 `json:"replicas,omitempty"`
+	// Secret reference to a Kubernetes Secret containing the admin portal
+	// endpoint URL. The Secret must be located in the same namespace.
 	// +optional
 	AdminPortalCredentialsRef *v1.LocalObjectReference `json:"adminPortalCredentialsRef,omitempty"`
+	// Secret reference to a Kubernetes secret containing the gateway
+	// configuration. The Secret must be located in the same namespace.
 	// +optional
 	EmbeddedConfigurationSecretRef *v1.LocalObjectReference `json:"embeddedConfigurationSecretRef,omitempty"`
+	// Kubernetes Service Account name to be used for the APIcast Deployment. The
+	// Service Account must exist beforehand.
 	// +optional
 	ServiceAccount *string `json:"serviceAccount,omitempty"`
+	// Image allows overriding the default APIcast gateway container image.
+	// This setting should only be used for dev/testing purposes. Setting
+	// this disables automated upgrades of the image.
 	// +optional
 	Image *string `json:"image,omitempty"`
+	// ExposedHost is the domain name used for external access. By default no
+	// external access is configured.
 	// +optional
 	ExposedHost *APIcastExposedHost `json:"exposedHost,omitempty"`
+	// DeploymentEnvironment is the environment for which the configuration will
+	// be downloaded from 3scale (Staging or Production), when using new APIcast.
+	// The value will also be used in the header X-3scale-User-Agent in the
+	// authorize/report requests made to 3scale Service Management API. It is
+	// used by 3scale just for statistics.
 	// +optional
 	DeploymentEnvironment *DeploymentEnvironmentType `json:"deploymentEnvironment,omitempty"` // THREESCALE_DEPLOYMENT_ENV
+	// DNSResolverAddress can be used to specify a custom DNS resolver address
+	// to be used by OpenResty.
 	// +optional
 	DNSResolverAddress *string `json:"dnsResolverAddress,omitempty"` // RESOLVER
+	// EnabledServices can be used to specify a list of service IDs used to
+	// filter the configured services.
 	// +optional
 	EnabledServices []string `json:"enabledServices,omitempty"` // APICAST_SERVICES_LIST
+	// ConfigurationLoadMode can be used to set APIcast's configuration load mode.
 	// +optional
 	// +kubebuilder:validation:Enum=boot;lazy
 	ConfigurationLoadMode *string `json:"configurationLoadMode,omitempty"` // APICAST_CONFIGURATION_LOADER
+	// LogLevel controls the log level of APIcast's OpenResty logs.
 	// +optional
 	// +kubebuilder:validation:Enum=debug;info;notice;warn;error;crit;alert;emerg
 	LogLevel *string `json:"logLevel,omitempty"` // APICAST_LOG_LEVEL
+	// PathRoutingEnabled can be used to enable APIcast's path-based routing
+	// in addition to to the default host-based routing.
 	// +optional
 	PathRoutingEnabled *bool `json:"pathRoutingEnabled,omitempty"` // APICAST_PATH_ROUTING
+	// ResponseCodesIncluded can be set to log the response codes of the responses
+	// returned by the API backend in 3scale.
```

Comment:
Not sure what the "API backend in 3scale" refers to: 3scale API?

Reply:
I'm not sure either. This description (and most of them) have been extracted from the source of truth which is APIcast's documentation. For this one specifically, it is https://github.com/3scale/APIcast/blob/master/doc/parameters.md#apicast_response_codes

Reply:
@eloycoto do you know what is meant exactly?

Reply:
or maybe @davidor?

Reply:
When that is enabled, APIcast sends the response code returned by the upstream API to apisonator, so you can see those response codes in the system UI.